src/routes/api.js

```diff
@@ -1,9 +1,23 @@
 import React from "react"
 import { ThreeColumnLayout } from "../components/three-column-layout"
+import useApiDocs from "../hooks/use-api-docs"
 import { useRouter } from "../hooks/use-router"
 
-export default function DocsPage(props) {
+export default function Api(props) {
+  let { publicClasses } = useApiDocs()
+
   let router = useRouter()
+  let classRouter = router.routerFor("/api/classes")
+
+  publicClasses
+    .filter(publicClass => !classRouter.has({ name: publicClass.name }))
+    .forEach(publicClass =>
+      classRouter.add({
+        name: publicClass.name,
+        label: publicClass.name,
+        path: `/${publicClass.slug}`,
+      })
+    )
```

Comment:
Yeah looking at this, I think ideally we do it all at build time so by the time we get here we aren't adding/removing routes. `useRouter` could `useStaticQuery` to get all known routes and setup the router once the first time it's used.